src/redux/polygonExplorer.js

```diff
@@ -0,0 +1,404 @@
+            status: 'ok',
+          },
+          payload: { assets },
+        },
+        true
+      )
+    );
+  };
+
+  const fetchAssetsBalancesAndPrices = async () => {
+    const chainId = ethereumUtils.getChainIdFromNetwork(network);
+    const assets = await getAssetsFromCovalent(
+      chainId,
+      accountAddress,
+      AssetTypes.polygon,
+      formattedNativeCurrency,
+      coingeckoIds,
+      allAssets,
+      genericAssets
+    );
+
+    if (assets === null) {
+      logger.debug('COVALENT FAILED, FALLING BACK');
+      fetchAssetsBalancesAndPricesFallback();
+      return;
+    }
+
+    if (!assets || !assets.length) {
+      const polygonExplorerBalancesHandle = setTimeout(
+        fetchAssetsBalancesAndPrices,
+        UPDATE_BALANCE_AND_PRICE_FREQUENCY
+      );
+      dispatch({
+        payload: {
+          polygonExplorerBalancesHandle,
+        },
+        type: POLYGON_EXPLORER_SET_BALANCE_HANDLER,
+      });
+      return;
+    }
+
+    const tokenAddresses = assets.map(
+      ({ asset: { asset_code } }) => asset_code
+    );
+
+    dispatch(emitAssetRequest(tokenAddresses));
+    dispatch(emitChartsRequest(tokenAddresses));
+
+    const prices = await fetchAssetPrices(
+      assets.map(({ asset: { coingecko_id } }) => coingecko_id),
+      formattedNativeCurrency
+    );
+
+    if (prices) {
+      Object.keys(prices).forEach(key => {
+        for (let i = 0; i < assets.length; i++) {
+          if (toLower(assets[i].asset.coingecko_id) === toLower(key)) {
+            if (!assets[i].asset.price.relative_change_24h) {
+              assets[i].asset.price.relative_change_24h =
+                prices[key][`${formattedNativeCurrency}_24h_change`];
+            }
+            break;
+          }
+        }
+      });
+    }
+
+    dispatch(
+      addressAssetsReceived(
+        {
+          meta: {
+            address: accountAddress,
+            currency: nativeCurrency,
+            status: 'ok',
+          },
+          payload: { assets },
+        },
+        true
+      )
+    );
+
+    const polygonExplorerBalancesHandle = setTimeout(
+      fetchAssetsBalancesAndPrices,
+      UPDATE_BALANCE_AND_PRICE_FREQUENCY
+    );
+    let polygonExplorerAssetsHandle = null;
+
+    dispatch({
+      payload: {
+        polygonExplorerAssetsHandle,
+        polygonExplorerBalancesHandle,
+      },
+      type: POLYGON_EXPLORER_SET_HANDLERS,
+    });
+  };
+  fetchAssetsBalancesAndPrices();
+};
+
+export const polygonExplorerClearState = () => (dispatch, getState) => {
+  const {
+    polygonExplorerBalancesHandle,
+    polygonExplorerAssetsHandle,
+  } = getState().polygonExplorer;
+
+  polygonExplorerBalancesHandle && clearTimeout(polygonExplorerBalancesHandle);
+  polygonExplorerAssetsHandle && clearTimeout(polygonExplorerAssetsHandle);
+  dispatch({ type: POLYGON_EXPLORER_CLEAR_STATE });
+};
+
+// -- Reducer ----------------------------------------- //
+const INITIAL_STATE = {
+  assetsFound: [],
```

Comment:
The state `assetsFound` is not used (same in the optimism and arbitrum explorer)